app/models/miq_server/worker_management/monitor.rb

```diff
@@ -52,16 +52,12 @@ def cleanup_failed_workers
   end
 
   def clean_worker_records
-    worker_deleted = false
-    miq_workers.each do |w|
-      next unless w.is_stopped?
+    workers_to_delete = miq_workers.select(:is_stopped?)
+    return if workers_to_delete.empty?
+
+    remove_workers(miq_workers.to_a.select(&:is_stopped?)) do |w|
       _log.info("SQL Record for #{w.format_full_log_msg}, Status: [#{w.status}] is being deleted")
-      worker_delete(w.pid)
```

Comment:
But then it doesn't delete any?

Reply:
`remove_workers` does that for us.
I debated removing this logging but this PR was aiming for a refactor only. (at least so far)